src/utils.jl

```diff
@@ -259,7 +259,34 @@ isshortdef(ex) = (@capture(ex, (fcall_ = body_)) &&
 
 function longdef1(ex)
   if @capture(ex, (arg_ -> body_))
-    Expr(:function, arg isa Symbol ? :($arg,) : arg, body)
+
+    if isexpr(arg, :tuple) && length(arg.args) == 1 && isexpr(arg.args[1], :parameters)
+      # Special case (; kws...) ->
+      fcall = Expr(:tuple, arg.args[1])
+
+      Expr(:function, fcall, body)
+    elseif isexpr(arg, :block) && any(a -> isexpr(a, :...) || isexpr(a, :(=)) || isexpr(a, :kw), arg.args)
+      # Has keywords in a block
```

Comment:
Sorry, I don't understand what this `elseif` branch is doing. What are "keywords in a block"? Can you give an example?

Reply:
Sure; take a look at the AST of this closure: `dump(quote; (a::Int; b=2) -> a+b; end)`